### PR TITLE
feat: export extra animation options for Pager

### PR DESCRIPTION
**Pull request by osdnk**
### Motivation

Sometimes I suppose a user might want to define own behavior of timing of spring and I want to make it possible with this PR

### Test plan

Add some config (e.g. `extraSpringConfig`) to `Pager`.

### CODE REVIEW

File: src/Pager.js [hunk 1]
```diff
@@ -96,6 +96,8 @@ const TIMING_CONFIG = {
 export default class Pager<T: Route> extends React.Component<Props<T>> {
   static defaultProps = {
     swipeVelocityThreshold: 1200,
+    extraSpringConfig: {},
```
Comment (satya164) [inline src/Pager.js]: Let's rename these to `springConfig` and `timingConfig` and then make them updatable when you change props.

File: src/TabView.js [hunk 2]
```diff
@@ -94,6 +94,8 @@ export default class TabView<T: Route> extends React.Component<
       swipeEnabled,
       swipeDistanceThreshold,
       swipeVelocityThreshold,
+      timingConfig,
```
Comment (osdnk) [inline src/TabView.js]: 👀

File: src/types.js [hunk 3]
```diff
@@ -41,4 +41,14 @@ export type PagerCommonProps = {|
   swipeVelocityThreshold?: number,
   onSwipeStart?: () => mixed,
   onSwipeEnd?: () => mixed,
+  springConfig: {|
```
Comment (osdnk) [inline src/types.js]: I think it's possible to reanimated nodes here

Comment (osdnk) [reply src/types.js]: No, but not now xD

File: src/Pager.js [hunk 4]
```diff
@@ -190,6 +202,45 @@ export default class Pager<T: Route> extends React.Component<Props<T>> {
...
+      this.props.springConfig.restDisplacementThreshold !== undefined
+        ? this.props.springConfig.restDisplacementThreshold
+        : 0.001
+    ),
+    overshootClamping: true,
+  };
+
+  _timingConfig = {
+    duration: new Value(
+      typeof this.props.timingConfig.duration !== 'undefined'
```
Comment (osdnk) [inline src/Pager.js]: @satya164

**Pull request merged**